### ISSUE TITLE
Investigate slow RetinaNet tests on CPU

### ISSUE THREAD

**Issue by oke-aditya** [Quick-AI/quickvision#64]
## 🐛 Bug

No Idea why our CI bottlenecks at RetinaNet tests.
Rest all work quite fast and we can finish CI testing in under 5 mins.

**To Reproduce**

Run tests using pytest
`pytest tests/test_retina.py` optionally use kkv flag and diagnose

### Expected behavior

Probably should work under 2 mins, it takes too much time.

### Environment

 - PyTorch Version (e.g., 1.0): 1.7
 - OS (e.g., Linux): Ubuntu 20.04
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source): 
 - Python version: 3.6, 3.7, 3.8
 - CUDA/cuDNN version: **None**
 - GPU models and configuration: CPU tests are slow

### Additional context

Not a huge issue but lets keep it open.

**Comment by zhiqwang** [comment]
This is weird. But I think set `model = retinanet.create_vision_retinanet(pretrained=True)` maybe helped.

**Comment by oke-aditya** [comment]
Actually RetinaNet from torchvision is slow. Lets investigate it bit more, and compare to FRCNN.

**Comment by oke-aditya** [comment]
RetinaNet Timings CUDA Indicates tests were on GPU.

```
307.79s call     tests/test_retinanet.py::EngineTester::test_fit
221.03s call     tests/test_retinanet.py::EngineTester::test_train_step_fpn
179.73s call     tests/test_retinanet.py::EngineTester::test_sanity_fit
89.37s call     tests/test_retinanet.py::EngineTester::test_train_sanity_fit
82.92s call     tests/test_retinanet.py::EngineTester::test_val_sanity_fit
70.28s call     tests/test_retinanet.py::LightningTester::test_lit_cnn_cuda
69.88s call     tests/test_retinanet.py::LightningTester::test_lit_retinanet_fpn
66.50s call     tests/test_retinanet.py::EngineTester::test_val_step_fpn
12.82s call     tests/test_retinanet.py::EngineTester::test_fit_cuda
11.00s call     tests/test_retinanet.py::EngineTester::test_infer
9.68s call     tests/test_retinanet.py::EngineTester::test_train
```

As we see `test_fit_cuda` is super fast whereas on CPU its super slow.

**Comment by zhiqwang** [comment]
It seems retinanet in torchvision 0.8 release doesn't use `batched_nms`, maybe it's the reason of this phenomenon.

**Comment by oke-aditya** [comment]
I'm unsure of 0.8.xrelease what they did, but on master they are using [batched_nms](https://github.com/pytorch/vision/blob/master/torchvision/models/detection/retinanet.py#L446)

**Comment by oke-aditya** [comment]
Let's dig into this a bit more and then raise an issue with torchvision team, maybe some operation is not vectorized or is not batched.

**Comment by zhiqwang** [comment]
The master branch has support batched_nms, the released 0.8.1 and 0.8.2 is using the [nms](https://github.com/pytorch/vision/blob/2f40a483d73018ae6e1488a484c5927f2b309969/torchvision/models/detection/retinanet.py#L456)

**Comment by oke-aditya** [comment]
Great catch 0.8.2 (latest release) too uses [nms](https://github.com/pytorch/vision/blob/v0.8.2/torchvision/models/detection/retinanet.py).
So possibly this will be fixed with torchvision 0.9.0
We can then verify by running stuff once to confirm

**Comment by oke-aditya** [comment]
I could see improvement in tests and I cross verified that release does contain a faster batches nms. So we should fine in closing this ! Thanks a lot @zhiqwang
